pallets/uniques/src/lib.rs

```diff
@@ -0,0 +1,849 @@
+				.ok_or(Error::<T, I>::UnknownCollectionOwner)?;
+			for (item, extra_deposit) in ItemExtraDeposits::<T, I>::drain_prefix(collection) {
+				let item_owner =
+					pallet_uniques::Pallet::<T, I>::owner(collection, item).ok_or(Error::<T, I>::UnknownItemOwner)?;
+				<T as pallet_uniques::Config<I>>::Currency::unreserve(&collection_owner, extra_deposit);
+				<T as pallet_uniques::Config<I>>::Currency::transfer(
+					&collection_owner,
+					&item_owner,
+					extra_deposit,
+					ExistenceRequirement::AllowDeath,
+				)?;
+			}
+			pallet_uniques::Pallet::<T, I>::destroy(origin, collection, witness)
+		}
+
+		/// Mint an item of a particular collection.
+		///
+		/// The origin must be Signed and the sender must be the Issuer of the `collection`.
+		///
+		/// - `collection`: The collection of the item to be minted.
+		/// - `item`: The item value of the item to be minted.
+		/// - `beneficiary`: The initial owner of the minted item.
+		///
+		/// Emits `Issued` event when successful.
+		///
+		/// Weight: `O(1)`
+		#[pallet::call_index(3)]
+		#[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::mint())]
+		pub fn mint(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			item: T::ItemId,
+			owner: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			pallet_uniques::Pallet::<T, I>::mint(origin, collection, item, owner)
+		}
+
+		/// Destroy a single item.
+		///
+		/// Origin must be Signed and the signing account must be either:
+		/// - the Admin of the `collection`;
+		/// - the Owner of the `item`;
+		///
+		/// - `collection`: The collection of the item to be burned.
+		/// - `item`: The item of the item to be burned.
+		/// - `check_owner`: If `Some` then the operation will fail with `WrongOwner` unless the
+		///   item is owned by this value.
+		///
+		/// Emits `Burned` with the actual amount burned.
+		///
+		/// Weight: `O(1)`
+		/// Modes: `check_owner.is_some()`.
+		#[pallet::call_index(4)]
+		#[pallet::weight(<T as Config<I>>::WeightInfo::burn())]
+		pub fn burn(
+			origin: OriginFor<T>,
+			collection: T::CollectionId,
+			item: T::ItemId,
+			check_owner: Option<AccountIdLookupOf<T>>,
+		) -> DispatchResult {
+			let collection_owner = pallet_uniques::Pallet::<T, I>::collection_owner(collection)
+				.ok_or(Error::<T, I>::UnknownCollectionOwner)?;
+			let item_owner =
+				pallet_uniques::Pallet::<T, I>::owner(collection, item).ok_or(Error::<T, I>::UnknownItemOwner)?;
+			pallet_uniques::Pallet::<T, I>::burn(origin, collection, item, check_owner)?;
+			let extra_deposit = ItemExtraDeposits::<T, I>::take(collection, item).unwrap_or_else(Zero::zero);
+			if !extra_deposit.is_zero() {
+				<T as pallet_uniques::Config<I>>::Currency::unreserve(&collection_owner, extra_deposit);
+				<T as pallet_uniques::Config<I>>::Currency::transfer(
+					&collection_owner,
+					&item_owner,
+					extra_deposit,
+					ExistenceRequirement::AllowDeath,
+				)
+			} else {
+				Ok(())
+			}
```

Comment:
Shouldn't this decrease the `total` as well?

Reply:
No it shouldn't. Because it again messes with the maximum that the collection owner allows to be spent on that collection. If someone burns a token they receive its intrinsic value.